### ISSUE TITLE
Native lib fatal error

### ISSUE THREAD

**Issue by jun1orose** [kohlschutter/junixsocket#103]
**Describe the bug**

I used `AFUNIXServerSocketChannel` and `Selector` api to accept new connection from client, but after new connection accepted and `SelectionKey.OP_READ` registered , next call to `Selector#select()` ends with fatal error.  

**To Reproduce**
```java
AFUNIXSelectorProvider provider = AFUNIXSelectorProvider.provider();
try (AFUNIXServerSocketChannel server = provider.openServerSocketChannel()) {
    server.bind(new AFUNIXSocketAddress(new File("/tmp/test.sock")));
    server.configureBlocking(false);
        
    Selector selector = provider.openSelector();
    server.register(selector, SelectionKey.OP_ACCEPT);

    ByteBuffer buffer = ByteBuffer.allocate(512);
        
    while (!Thread.currentThread().isInterrupted()) {
        selector.select();

        Set<SelectionKey> keys = selector.selectedKeys();
        Iterator<SelectionKey> it = keys.iterator();

         while (it.hasNext()) {
             SelectionKey key = it.next();

             if (key.isAcceptable()) {
                 AFUNIXSocketChannel channel = server.accept();

                 System.out.println("[Server] Accept. " + channel);

                 channel.configureBlocking(false);
                 channel.register(selector, SelectionKey.OP_READ);
             }

             if (key.isReadable()) {
                 AFUNIXSocketChannel channel = (AFUNIXSocketChannel) key.channel();
                 channel.read(buffer);
                 System.out.println("[Server] Read: " + new String(buffer.array()));
                 buffer.clear();
             }
             
             it.remove()

        }
    }
}
```

**Expected behavior**

Next `Selector#select()`call blocking until new key selected.

**Output/Screenshots**

![image](https://user-images.githubusercontent.com/29614719/161301206-ebb08782-84a3-4688-b43b-f3ec9447b27d.png)

**Environment (please complete the following information):**
 - OS: Linux
 - Manjaro
 - Version 21.2.3
 
 
Selftest results:
PASS	junixsocket-common	101/105 (4 skipped)
PASS	junixsocket-rmi	7/7

Supported capabilities:   [CAPABILITY_PEER_CREDENTIALS, CAPABILITY_ANCILLARY_MESSAGES, CAPABILITY_FILE_DESCRIPTORS, CAPABILITY_ABSTRACT_NAMESPACE, CAPABILITY_DATAGRAMS, CAPABILITY_NATIVE_SOCKETPAIR]
Unsupported capabilities: []

Selftest PASSED

**Notes**
[hs_err_pid20917.log](https://github.com/kohlschutter/junixsocket/files/8399383/hs_err_pid20917.log)

**Comment by kohlschuetter** [comment]
This should be fixed in 2.5.0 / commit 2f8f096b84fba1ae26058d41cc03df47719379df. Thank you for reporting!

**Comment by kohlschuetter** [comment]
@jun1orose  Assuming this is working for you now. Please reopen if you encounter any issues with version 2.5.0 or newer. Thanks again for reporting!
